middleware/grpc.go

```diff
@@ -0,0 +1,104 @@
+package middleware
+
+import (
+	"context"
+	"io"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	grpcUtils "github.com/weaveworks/common/grpc"
+	"github.com/weaveworks/common/httpgrpc"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// PrometheusGRPCUnaryInstrumentation records duration of gRPC requests client side.
+func PrometheusGRPCUnaryInstrumentation(metric *prometheus.HistogramVec) grpc.UnaryClientInterceptor {
```

Comment:
The equivalent server call in this package is `UnaryServerInstrumentInterceptor`, so perhaps `UnaryClientInstrumentInterceptor`?

Reply:
@bboreham Thanks, done.